From b476faf1c4f96c093ea1a8a0c824dea9f55b665f Mon Sep 17 00:00:00 2001
From: Christian Hesse <list@eworm.de>
Date: Mon, 25 Sep 2017 17:36:24 +0100
Subject: [PATCH] Do not include stdio.h before dnsmasq.h

We define some constants in dnsmasq.h, which have an influence on
stdio.h. So do not include stdio.h before dnsmasq.h.
---
 src/dnsmasq.h | 6 ++++++
 src/helper.c  | 1 -
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/dnsmasq.h b/src/dnsmasq.h
index 7a18898..421488b 100644
--- a/src/dnsmasq.h
+++ b/src/dnsmasq.h
@@ -16,6 +16,12 @@
 
 #define COPYRIGHT "Copyright (c) 2000-2016 Simon Kelley"
 
+/* We do defines that influence behavior of stdio.h, so complain
+   if included too early. */
+#ifdef _STDIO_H
+#  error "Header file stdio.h included too early!"
+#endif 
+
 #ifndef NO_LARGEFILE
 /* Ensure we can use files >2GB (log files may grow this big) */
 #  define _LARGEFILE_SOURCE 1
diff --git a/src/helper.c b/src/helper.c
index 4fffa27..68ce9a7 100644
--- a/src/helper.c
+++ b/src/helper.c
@@ -14,7 +14,6 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include <stdio.h>
 #include "dnsmasq.h"
 
 #ifdef HAVE_SCRIPT
-- 
2.9.5

